File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int l=0,mini=prices[0],maxi=0;
        for(int i=0;i<n;i++){
      
                maxi = max(maxi, prices[i]-mini);
                mini = min(mini, prices[i]);

            
        }
        return maxi;
    }
};

File: Data Structures & Algorithms/task-scheduling/submission-0.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        freq = [0]*26
        for i in range(len(tasks)):
            freq[ord(tasks[i])-ord('A')]+=1
        freq.sort()
        maxi = freq[-1]
        gap = (maxi-1)*n

        for i in range(24,-1,-1):
            gap-=min(maxi-1,freq[i])
        return max(0,gap)+len(tasks)